lib/snapshot-manager.js

```diff
@@ -449,12 +449,49 @@ const determineSnapshotDir = mem(({file, fixedLocation, projectDir}) => {
 
 exports.determineSnapshotDir = determineSnapshotDir;
 
-function load({file, fixedLocation, projectDir, recordNewSnapshots, updating}) {
+function determineSnapshotPaths({file, fixedLocation, projectDir}) {
 	const dir = determineSnapshotDir({file, fixedLocation, projectDir});
 	const relFile = path.relative(projectDir, resolveSourceFile(file));
 	const name = path.basename(relFile);
 	const reportFile = `${name}.md`;
 	const snapFile = `${name}.snap`;
+
+	return {
+		dir,
+		relFile,
+		snapFile,
+		reportFile
+	};
+}
+
+// Remove snapshot and report if they exist. Returns an array containing the
+// paths of the touched files.
+async function cleanSnapshots({file, fixedLocation, projectDir}) {
+	const {dir, snapFile, reportFile} = determineSnapshotPaths({file, fixedLocation, projectDir});
+
+	async function clean(file) {
+		try {
+			await fs.promises.unlink(file);
+			return [file];
+		} catch (error) {
+			if (error.code === 'ENOENT') {
+				return [];
+			}
+
+			throw error;
+		}
+	}
+
+	return [
+		...await clean(path.join(dir, snapFile)),
+		...await clean(path.join(dir, reportFile))
```

Comment:
This might all be a bit simpler if it was synchronous, what do you think? It's not going to matter performance wise.

Reply:
My personal preference is to write async code whenever there's an underlying async api/operation, on the grounds that 1) the difference is usually just some `async/await`s, if the sync version looks much simpler it's probably hiding something and 2) spec/performance can always change in the future, and changing sync code to async propagates through every sync caller while changing async code to sync is usually very simple.

In this case, sync code would conform better to the rest of the `snapshot-manager` api, but then I'm not sure _why_ it's all sync, maybe it was written pre-`fs.promises`?

Reply:
Fair enough.

> In this case, sync code would conform better to the rest of the `snapshot-manager` api, but then I'm not sure _why_ it's all sync, maybe it was written pre-`fs.promises`?

Pre-`async`/`await` too 😉 

I have a mild preference for keeping this consistent with the existing logic.